test/splitter.js

```diff
@@ -0,0 +1,160 @@
+                    ev.remainder.toString(10) === "1";
+        }, 'TestEvent should be emitted with correct parameters');
+    })
+
+    it("Transaction reverts if the deposit amount is zero", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(RecipientTwoAddress, RecipientOneAddress,{from: senderAddress, value: 0}),
+            "The value must be greater than 0"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it("Transaction reverts if the first recipient is the same as the second recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(RecipientOneAddress, RecipientOneAddress,{from: senderAddress, value: 4}),
+            "The first recipient is the same as the second recipient"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it("Transaction reverts if the sender's address is also a recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(senderAddress, RecipientTwoAddress, {from: senderAddress, value: 4}),
+            "The sender cannot also be a recipient"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it('Second recipient can successfully withdraw 2 wei', async () => {
+
+        const depositAmount = web3.utils.toWei(web3.utils.toBN(1), "ether");
+        const withDrawAmount = web3.utils.toBN(2);
+
+        // Take snapshot of the recipients ETH balance
+        let initRecipientTwoEthBalance = await web3.eth.getBalance(RecipientTwoAddress);
+        initRecipientTwoEthBalance = web3.utils.toBN(initRecipientTwoEthBalance);
+
+        // Deposit into the contract
+        await splitter.splitDeposit(RecipientOneAddress, RecipientTwoAddress, {from: senderAddress, value: depositAmount});
+
+        // Take snapshot of the recipients new contract balance
+        let initRecipientTwoContractBalance = await splitter.balances(RecipientTwoAddress);
+        initRecipientTwoContractBalance = web3.utils.toBN(initRecipientTwoContractBalance)
+
+        // Recipient withdraws
+        const receipt = await splitter.withdraw(withDrawAmount, {from: RecipientTwoAddress});
+        const cost = await gasCost(receipt);
+
+        // Get the recipient's new ETH and contract balances
+        const carolEthBalance = await web3.eth.getBalance(RecipientTwoAddress);
+        const carolContractBalance = await splitter.balances(RecipientTwoAddress);
+
+        // Calculate the expected new ETH and contract balances
+        const expectedRecipientTwoEthBalance = initRecipientTwoEthBalance.sub(cost).add(withDrawAmount);
+        const expectedRecipientTwoContractBalance = initRecipientTwoContractBalance.sub(withDrawAmount);
+
+        assert.strictEqual(carolEthBalance.toString(10), expectedRecipientTwoEthBalance.toString(10));
+        assert.strictEqual(carolContractBalance.toString(10), expectedRecipientTwoContractBalance.toString(10));
```

Comment:
`, "0");` ?

Reply:
Also, you test that the "owed" info has been correctly updated in the contract storage.
What you need to check also is that the money actually landed in the recipient's `web.eth.getBalance` balance.

Reply:
> `, "0");` ?

I am sorry. I do not understand this comment.

Reply:
At first I thought you would withdraw everything, where `expectedRecipientTwoContractBalance` would be `0`. But I see that it is not the case.

Reply:
Sorry. I subsequently understood what you were referring to.

Your expectation was right. initRecipientTwoOwed.sub(withDrawAmount) should be "0".

Initially 2 is added the balance and then 2 is withdrawn. Should I be more explicit and use "0" rather than calculate it using:

`initRecipientTwoOwed.sub(withDrawAmount)` ?